packages/get-platform/src/getPlatform.ts

```diff
@@ -200,7 +200,21 @@ export async function getSSLVersion(args: GetOpenSSLVersionParams): Promise<GetO
 
   debug('Falling back to "ldconfig" and other generic paths')
   const libsslFilename: string | undefined = await getFirstSuccessfulExec([
-    'ldconfig -p | sed "s/.*=>s*//" | sed "s/.*///" | grep ssl | sort',
+    /**
+     * The `ldconfig -p` returns the dynamic linker cache paths, where libssl.so files are likely to be included.
+     * Each line looks like this:
+     * 	libssl.so (libc6,hard-float) => /usr/lib/arm-linux-gnueabihf/libssl.so.1.1
+     * But we're only interested in the filename, so we use sed to remove everything before the `=>` separator,
+     * and then we remove the path and keep only the filename.
+     * The second sed commands uses `|` as a separator because the paths may contain `/`, which would result in the
+     * `unknown option to 's'` error (see https://stackoverflow.com/a/9366940/6174476) - which would silently
+     * fail with error code 0.
+     */
+    'ldconfig -p | sed "s/.*=>s*//" | sed "s|.*/||" | grep ssl | sort',
```

Comment:
Here's how to test this:
- Open a debian armv7l shell with `docker run --platform="linux/arm" --rm -ti --entrypoint=/bin/sh node:lts-slim`
- Run the original command: `ldconfig -p | sed "s/.*=>s*//" | sed "s/.*///" | grep ssl | sort`
- You should see the message "sed: -e expression #1, char 7: unknown option to `s"
- Run `echo $?` to check the status code, which should be `0`
- Now run the updated command: `ldconfig -p | sed "s/.*=>s*//" | sed "s|.*/||" | grep ssl | sort`
- The result should be `libssl.so.1.1`

<img width="637" alt="Screenshot 2023-01-15 at 11 51 49" src="https://user-images.githubusercontent.com/12381818/212536506-b1e28c6e-9499-4ee2-8e8c-2912d58cc595.png">